src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterReceiver.cs

```diff
@@ -96,7 +135,38 @@ public Task MessagesDeliveredAsync(IList<IBatchContainer> messages)
 
         private IList<IBatchContainer> BuildBatchList(IEnumerable<EventData> messages)
         {
-            return messages.Select(EventHubBatchContainer.FromEventData).ToList();
+            return messages.Select(CacheAndCreateStreamActivityNotificationBatch).ToList();
```

Comment:
Replace with loop.  State is being modified, so linq can introduce maintenance issues.